astropy/units/__init__.py

```diff
@@ -10,44 +10,56 @@
 code under a BSD license.
 """
 
-# Import order matters here -- circular dependencies abound!
-# Lots of things to import - go from more basic to advanced, so that
-# whatever advanced ones need generally has been imported already;
-# this also makes it easier to understand where most time is spent
-# (e.g., using python -X importtime).
-
-from .core import set_enabled_units
-
-# isort: off
-from .core import *
-from .quantity import *
-
-from . import astrophys, cgs, misc, photometric, si
-from .function import units as function_units
-
-from .si import *
+from . import (
+    astrophys,
+    cgs,
+    core,
+    decorators,
+    misc,
+    photometric,
+    physical,
+    quantity,
+    si,
+    structured,
+)
 from .astrophys import *
-from .photometric import *
 from .cgs import *
-from .physical import *
-from .function.units import *
+from .core import *
+from .core import set_enabled_units
+from .decorators import *
 from .misc import *
+from .photometric import *
+from .physical import *
+from .quantity import *
+from .si import *
+from .structured import *
 
+# Import order matters here -- circular dependencies abound!
+# isort: split
+from . import equivalencies
 from .equivalencies import *
-
+from .function import units as function_units
 from .function.core import *
 from .function.logarithmic import *
+from .function.units import *
 
-from .decorators import *
-from .structured import *
-
-# isort: on
-
-del bases
+__all__ = []
+__all__ += core.__all__
+__all__ += quantity.__all__
+__all__ += decorators.__all__
+__all__ += structured.__all__
+__all__ += si.__all__
+__all__ += cgs.__all__
+__all__ += astrophys.__all__
+__all__ += physical.__all__
+__all__ += misc.__all__
+__all__ += photometric.__all__
+__all__ += structured.__all__
+__all__ += equivalencies.__all__
+__all__ += function_units.__all__
```

Comment:
Why not write it as e.g.:

```
__all__ = (
    module1.__all__ +
    module2.__all__ +
    ...
)
```

rather than do it via a series of updates?

Reply:
See https://github.com/python/typing/blob/main/docs/source/libraries.rst#library-interface-public-and-private-symbols for how `__all__` can be updated and make static type checkers happy.

Reply:
``__all__ += submodule.__all__`` is my personal preference in this scenario. We could do `__all__.extend(submodule.__all__)`, but I find it less clean.